### PR TITLE
[release/6.3] Update hosted and BYOC agents used in our builds

### PR DESCRIPTION
**Pull request by dougbu**
- dotnet/aspnetcore-internal#3540

### CODE REVIEW

**Comment by dougbu** [comment]
**NO-MERGE** because the branches aren't open

File: azure-pipelines.yml [hunk 1]
```diff
@@ -13,7 +13,7 @@ variables:
 trigger:
   - master
   - release/*
-  - internal/release/*
+  - internal/release/6.*
```
Comment (Pilchie) [inline azure-pipelines.yml]: Why this change?

Comment (dougbu) [reply azure-pipelines.yml]: I added an extra commit in an attempt to figure out why the database cleanup was failing in the release branches but succeeded (I thought) in 'master'. Since the 'master' validation build now failed, I suspect something else is going on -- perhaps LocalDb is slightly different or missing on these agents.

Comment (dougbu) [reply azure-pipelines.yml]: Will remove that commit once I understand what's going on w/ the failing database cleanups.

Comment (dougbu) [reply azure-pipelines.yml]: The commit adding the unrelated changes is gone now. Much less to see here 😈

**Comment by dougbu** [comment]
Though removing the extra commit triggered another build, I doubt that will succeed. Suspect LocalDB issues in the Server agents -- as discussed in #1612.

**Comment by dougbu** [comment]
No need to update this branch
